catalog/transformer/catalog-transformer-overlay/src/main/java/ddf/catalog/transformer/OverlayActionProvider.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.transformer;
+
+import java.io.UnsupportedEncodingException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang.Validate;
+import org.codice.ddf.configuration.SystemBaseUrl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.vividsolutions.jts.geom.Geometry;
+
+import ddf.action.Action;
+import ddf.action.ActionProvider;
+import ddf.action.impl.ActionImpl;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.transform.CatalogTransformerException;
+
+public class OverlayActionProvider implements ActionProvider {
+    private static final Logger LOGGER = LoggerFactory.getLogger(OverlayActionProvider.class);
+
+    private static final String ID = "catalog.data.metacard.";
+
+    // TODO (jrnorth) - fix this
+    private static final String DESCRIPTION =
+            "Provides a metacard URL that transforms the metacard into a geographically aligned image (suitable for overlaying on a map) via the ";
+
+    private static final String TITLE = "Export as ";
+
+    private static final String OVERLAY_PREFIX = "overlay.";
+
+    private static final Pattern OVERLAY_TRANSFORMER_ID_PATTERN =
+            Pattern.compile("^overlay\\.\\w+");
```

Comment:
Instead of repeating "overlay.", how about: Pattern.compile("^" + Pattern.quote(OVERLAY_PREFIX) + "\\w+"):